aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -389,6 +390,21 @@ public class AwsProperties implements Serializable {
    */
   @Deprecated public static final boolean CLIENT_ENABLE_ETAG_CHECK_DEFAULT = false;
 
+  /**
+   * This flag controls whether the S3 client will be initialized during the S3FileIO
+   * initialization, instead of normal lazy initialization upon use. This is needed to deal with a
```

Comment:
I don't think we need to explain too much about the use case for LF, this is a generic S3FileIO config. Maybe something like "This is for cases that the credentials to use might change and needs to be pre-loaded"